### DISCUSSION TITLE
Re-enable root account in Cockpit?

### DISCUSSION THREAD

**Discussion by Verhoeckx** [cockpit-project/cockpit#16133] [Q&A] [answered]
Hello Cockpit developers,

Just out of curiosity: 
How do you re-enable the root account after you have disabled it?

I disabled to root account to work only from my default user account (with sudo rights). I wanted to check mark the option `Never lock account` for this account but I see that's not possible anymore (and probably also not necessary anymore)?

**Comment by Verhoeckx** [comment, marked as answer]
Oh wait, I found a button called `Limited access` with which I can get administrative access :smile: !
Now I can, if I want to, re-enable the root account. Problem solved!

I also found out that it's not necessary to click on `Never lock account`. It's a notification and not an action that I can take (which I thought it was). Stupid me :smile: !

**Reply by garrett** [reply]
Do you have any suggestions on what we could do to make this more obvious for you (and others)?

(There's also a banner on the overview page about this when you're in limited access mode.)

> Stupid me :smile: !

Nah, it just means we need to improve the software when it's not clear enough. It's not your fault; it's the software's. And we can make it better. :wink:

**Reply by Verhoeckx** [reply]
Hello @garrett

I think the location of switch account / change access button is fine but it's only not really clear that it's a button. Now it read (to me) as a statement: (you have) Limited access and (you have) Administrative access. 

When you press a button something happens and this is why a button should (always) have a verb that expresses the action. In this case the verb could be `switch` or `change` and the noun `account` or `access`. 

When you add a switch account / change access icon to it, it will be even more clear that you can switch accounts or change access. For example this one:
![user_exchange-512](https://user-images.githubusercontent.com/64807887/126688281-43d81a90-61c9-43b8-bfd6-0003349fdb9d.png)
or this one:
![2011004](https://user-images.githubusercontent.com/64807887/126688365-cb0dbda4-cb45-44e3-8b91-dfb3e4a1a1e2.png)

----

I think the banner in the Overview is also really good. I think I missed the button `Turn on administrative access` because it's aligned to the right side. If you align it to the left side (to the right of the text) I think it's more visible (but also somewhat less esthetic).

**Reply by Verhoeckx** [reply]
Mmm, I realize that the word `account` can be really confusing because then people might thing it's the same as logging in and logging out. You should probably stick to the noun `access`.

I think my idea of adding an icon with a person (account) on it is also not a good idea because this has the same problem as stated above. Maybe an icon that has something to do with rights?

**Reply by Verhoeckx** [reply]
Maybe just move the blue button `Turn on administrative access` to the top bar?

Rename `administrative` to `administrator`?

And when you have administrator rights: `Turn off administrator rights`?

Icon is not really necessary.

**Reply by garrett** [reply]
Thanks for your continued feedback!

I should look at this again and try to make it clearer.

The big issue is that there are some cases where people use Cockpit without administrative access (and will never be given full access), so we need to strike a balance. 

But I agree. It's not obvious enough and doesn't look so clickable. (This is the reason behind adding the banner. But, yeah, people might not see it and may proceed to other pages without realizing they're not in admin mode.)

**Reply by garrett** [reply]
I've mocked up some of these suggestions, like making the top look more button-like, adding an icon, and such @ https://github.com/cockpit-project/cockpit/issues/14305.

It's a WIP. I should probably implement some or all of this in a PR soon so we can see how it feels while using Cockpit.

**Reply by Verhoeckx** [reply]
I read your comment: all good points! Nice design!

**Comment by Verhoeckx** [comment]
I wonder however if it's possible to lock all accounts (so you won't be able to log in anymore in Cockpit)?
Because I still have the option to also lock my own user account :fearful: !

**Reply by Verhoeckx** [reply]
I still wonder if this is possible (locking the root and user account) but I'm not going to try it because I don't want to lock myself out of Cockpit.

**Reply by garrett** [reply]
This is a great point. I've opened an issue @ https://github.com/cockpit-project/cockpit/issues/16152

**Reply by Verhoeckx** [reply]
Great: I see that's it being working on!

**Reply by garrett** [reply]
We just landed the change in https://github.com/cockpit-project/cockpit/pull/16159. Preventing locking one's self out will be in the next Cockpit release. Thanks pointing this out! :+1:

**Reply by Verhoeckx** [reply]
My pleasure :wink: !
